tpu-provisioner/internal/cloud/gke.go

```diff
@@ -262,6 +262,13 @@ func (g *GKE) nodePoolForPod(name string, p *corev1.Pod) (*containerv1beta1.Node
 		LabelJobSetNamespace: p.Namespace,
 	}
 
+	// Copy configured labels from the Pod to the Node.
+	for _, key := range g.ClusterContext.PodToNodeLabels {
```

Comment:
Does it make sense to add this list in a pod label (say xx/pod-to-node-labels: label1,label2), so we can control what's being populated to node labels, by specifying what's in the list?

Reply:
Thats possible for sure! The config would probably fit better as an annotation. Will update PR.

Reply:
Done